app/models/repository.rb

```diff
@@ -64,8 +57,8 @@ def starred_by?(user)
   # digest.
   def groupped_tags
     tags.group_by(&:digest).values.sort do |x, y|
-      y.first.updated_at <=> x.first.updated_at
-    end
+      y.first.updated_at.to_f <=> x.first.updated_at.to_f
+    end.map(&:sort)
```

Comment:
Why are we sorting twice? Was this really causing an issue?

Reply:
We are not sorting twice, we are sorting differents elements 😁. With `group_by` you get an array of arrays. So, something like this:

```
[[tag0], [tag2, tag1]]
```

The `sort` on the block sorts on the `updated_at` element of the first element of each sub-array. So, it's: `[[tag0], [tag2, tag1]]` vs `[[tag2, tag1], [tag0]]`. 

The second sort is about the elements inside of each subarray. So  `[[tag2, tag1], [tag0]]` vs  `[[tag1, tag2], [tag0]]`.

MariaDB always sorted this second sort by name, but PostgreSQL does not make such assumptions, so they came with a different order.

Now, for example, let's say that we want to test that the deletion of tags works. The order matters, because the VCR cassette is recorded to first send a `delete` call with `tag1`, and then another one with `tag2`, for example, not the other way around.

Reply:
Gotcha. 👍